File: mediq-backend/src/main/java/com/mediq/service/ReferralService.java
package com.mediq.service;

import com.mediq.constants.ReferralStatus;
import com.mediq.dto.CreateReferralRequest;
import com.mediq.dto.ReferralResponse;
import com.mediq.dto.UpdateReferralStatusRequest;

import java.util.List;

public interface ReferralService {

    ReferralResponse createReferral(CreateReferralRequest request);

    ReferralResponse updateReferralStatus(Long referralDbId, UpdateReferralStatusRequest request);

    ReferralResponse getReferralById(Long referralDbId);

    ReferralResponse getReferralByCode(String referralId);

    ReferralResponse getReferralByConsultationId(Long consultationId);

    List<ReferralResponse> getPatientReferrals(Long patientId);

    List<ReferralResponse> getReferralsByStatus(Long campId, ReferralStatus status);

    byte[] generateReferralPdf(Long referralDbId);
}

File: mediq-frontend/src/services/referralService.js
import api from './api';

export const referralService = {
  createReferral: async (data) => api.post('/referrals', data),
  updateReferralStatus: async (id, status, remarks) => api.patch(`/referrals/${id}/status`, { status, remarks }),
  getReferralById: async (id) => api.get(`/referrals/${id}`),
  getReferralByCode: async (referralId) => api.get(`/referrals/code/${referralId}`),
  getReferralByConsultationId: async (consultationId) => api.get(`/referrals/consultation/${consultationId}`),
  getPatientReferrals: async (patientId) => api.get(`/referrals/patient/${patientId}`),
  getReferralsByStatus: async (campId, status) => api.get(`/referrals/status/${campId}`, { params: { status } }),
  getReferralPdfUrl: (id) => `/api/v1/referrals/${id}/pdf`,
  downloadReferralPdf: async (id) => api.get(`/referrals/${id}/pdf`, { responseType: 'blob' }),
};

export default referralService;
